taxonomy-editor-frontend/src/pages/go-to-project/index.tsx

```diff
@@ -8,34 +8,54 @@ import EditIcon from "@mui/icons-material/Edit";
 import useFetch from "../../components/useFetch";
 import { API_URL } from "../../constants";
 import { toSnakeCase, toTitleCase } from "../../utils";
+import type { ProjectsAPIResponse } from "../../backend-types/types";
 
-const GotoProject = ({ resetNavLinks }) => {
-  const {
-    data: incomingData,
-    isPending,
-    isError,
-    errorMessage,
-  } = useFetch(`${API_URL}projects?status=OPEN`);
-  const [projectData, setProjectData] = useState([]);
+type ProjectType = {
+  id: string;
+  projectName: string;
+  taxonomyName: string;
+  branchName: string;
+  description: string;
+  errors_count: number;
+};
+
+type Props = {
+  resetNavLinks: () => void;
+};
+
+const GoToProject = ({ resetNavLinks }: Props) => {
+  const [projectData, setProjectData] = useState<ProjectType[]>([]);
   const navigate = useNavigate();
 
+  const { data, isPending, isError } = useFetch<ProjectsAPIResponse>(
+    `${API_URL}projects?status=OPEN`
+  );
+
   useEffect(() => {
-    const renderedProjects = [];
-    if (incomingData) {
-      incomingData.forEach((element) => {
-        const projectNode = element[0];
-        renderedProjects.push({
-          id: Math.random().toString(), // Used as Material Table component key
-          projectName: projectNode["id"],
-          taxonomyName: toTitleCase(projectNode["taxonomy_name"]),
-          branchName: projectNode["branch_name"],
-          description: projectNode["description"],
-          errors_count: projectNode["errors_count"],
+    let newProjects: ProjectType[] = [];
+
+    if (data) {
+      const backendProjects = data
```

Comment:
Why do we need two arrays `backendProjects` and `newProjects` for setting the state of `projectData`? It seems to be repetitive.

Reply:
I think is easier to follow when you set a variable in one go (after the `.map`), instead of defining it inside a `.forEach`. This way you know the complete set of information that is being defining. On the other side, `newProjects` are the new value that the react state will take and `backendProjects` are the projects the backend is returning.